integrations/integrations.go

```diff
@@ -0,0 +1,201 @@
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L624
+func startApp(
+	svrCtx *server.Context,
+	appCreator servertypes.AppCreator,
+	opts server.StartCmdOptions,
+) (app servertypes.Application, cleanupFn func(), err error) {
+	traceWriter, traceCleanupFn, err := SetupTraceWriter(svrCtx.Logger, svrCtx.Viper.GetString(FlagTraceStore))
+	if err != nil {
+		return app, cleanupFn, err
+	}
+
+	home := svrCtx.Config.RootDir
+	db, err := opts.DBOpener(home, server.GetAppDBBackend(svrCtx.Viper))
+	if err != nil {
+		return app, cleanupFn, err
+	}
+
+	// TODO: Check if is testnet and implement `testnetify` function
+
+	app = appCreator(svrCtx.Logger, db, traceWriter, svrCtx.Viper)
+	cleanupFn = func() {
+		traceCleanupFn()
+		if localErr := app.Close(); localErr != nil {
+			svrCtx.Logger.Error(localErr.Error())
+		}
+	}
+
+	return app, cleanupFn, nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L307
+// We modify this function to start a Monomer node in-process instead of a Comet node.
+func startInProcess(
+	svrCtx *server.Context,
+	svrCfg serverconfig.Config,
+	clientCtx client.Context,
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	cmtCfg := svrCtx.Config
+	gRPCOnly := svrCtx.Viper.GetBool(FlagGRPCOnly)
+
+	g, ctx := GetCtx(svrCtx, true)
+
+	if gRPCOnly {
+		svrCtx.Logger.Info("starting node in gRPC only mode; Monomer is disabled")
+		svrCfg.GRPC.Enable = true
+	} else {
+		svrCtx.Logger.Info("starting Monomer node in-process")
+		// Start the Monomer node
+		_, monomerCleanupFn, err := startMonomerNode()
+		if err != nil {
+			return err
+		}
+		defer monomerCleanupFn()
+
+		// Add the tx service to the gRPC router. We only need to register this
+		// service if API or gRPC is enabled, and avoid doing so in the general
+		// case, because it spawns a new local CometBFT RPC client.
+		/* if svrCfg.API.Enable || svrCfg.GRPC.Enable {
+		    clientCtx = clientCtx.WithClient(local.New(tmNode))
+		    app.RegisterTxService(clientCtx)
+		    app.RegisterTendermintService(clientCtx)
+		    app.RegisterNodeService(clientCtx, svrCfg)
+		} */
+	}
+
+	grpcSrv, clientCtx, err := StartGrpcServer(ctx, g, svrCfg.GRPC, clientCtx, svrCtx, app)
+	if err != nil {
+		return err
+	}
+
+	err = StartAPIServer(ctx, g, svrCfg, clientCtx, svrCtx, app, cmtCfg.RootDir, grpcSrv)
+	if err != nil {
+		return err
+	}
+
+	if opts.PostSetup != nil {
+		if err := opts.PostSetup(svrCtx, clientCtx, ctx, g); err != nil {
+			return err
+		}
+	}
+
+	return g.Wait()
+}
+
+// Starts the Monomer node in-process in place of the Comet node.
+func startMonomerNode() (*node.Node, func(), error) {
+	chainID := monomer.ChainID(1)
+	engineWS, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, nil, err
+	}
+	cometListener, err := net.Listen("tcp", "127.0.0.1:0")
```

Comment:
We should obtain the addresses from the config. We may need to add some configurations (or just use environment variables if the Cosmos SDK doesn't give us hooks to add configs).

Reply:
Got it. Do we want the P2P address or the RPC?

Reply:
RPC. We don't do p2p right now.

Reply:
Addressed.